lib/logproto/logproto-proxied-text-server.c

```diff
@@ -68,7 +68,11 @@ _log_proto_proxied_text_server_handshake(LogProtoServer *s)
   status = self->fetch_from_buffer(&self->super.super.super, &msg, &msg_len, &may_read, NULL, NULL);
 
   if (status != LPS_SUCCESS)
-    return status;
+    {
+      if (status == LPS_AGAIN)
+        self->handshake_done = FALSE;
```

Comment:
I may be mistaken, but my understanding is that if we set `handshake_done` to TRUE only if we are really done with the handshake (we've extracted a full line that contains the proxy information or `PROXY UNKNOWN`), then `LPS_AGAIN` would not be needed here.

Reply:
it's not that simple - the method is called even when a TLS handshake is in progress, when this is the case we cannot set handhsake_done to TRUE

Reply:
one additional note: handshake_done is TRUE when we've received a string that should be parsed as a header
it is possible that the handshake will fail - in that case handshake_done is set to TRUE (and connection will be closed...)